docs/technologies.md

```diff
@@ -0,0 +1,423 @@
+''')
+
+create_ports_table = dataset.create_table('''
+    CREATE TABLE IF NOT EXISTS ports (
+      port_name STRING,
+      port_latitude FLOAT64,
+      port_longitude FLOAT64,
+      country STRING,
+      index_number STRING)
+''')
+
+select_polish_ports = dataset.write_truncate('ports', '''
+        SELECT port_name, port_latitude, port_longitude
+        FROM `{more_ports}`
+        WHERE country = 'POL'
+        ''', partitioned=False)
+
+populate_ports_table = dataset.collect('''
+        INSERT INTO `{more_ports}` (port_name, port_latitude, port_longitude, country, index_number)
+        VALUES 
+        ('GDYNIA', 54.533333, 18.55, 'POL', '28740'),
+        ('GDANSK', 54.35, 18.666667, 'POL', '28710'),
+        ('SANKT-PETERBURG', 59.933333, 30.3, 'RUS', '28370'),
+        ('TEXAS', 34.8, 31.3, 'USA', '28870');
+        ''')
+
+
+internationalports_workflow = Workflow(
+        workflow_id='internationalports',
+        definition=[
+                create_ports_table.to_job(id='create_ports_table'),
+                create_polish_ports_table.to_job(id='create_polish_ports_table'),
+                populate_ports_table.to_job(id='populate_ports_table'),
+                select_polish_ports.to_job(id='select_polish_ports'),
+        ],
+        schedule_interval='@once')
+```
+
+There are two notable elements in the `internationalports.workflow` module:
+
+* `DatasetConfig` class which defines a BigQuery dataset you want to interact with
+* `dataset: Dataset` object which allows you to perform various operations on a defined dataset
+
+Using a dataset object you can describe operations that you want to perform. Next, you arrange them into a workflow.
+
+Take a look at the example operation, which creates a new table:
+
+```python
+create_polish_ports_table = dataset.create_table('''
+    CREATE TABLE IF NOT EXISTS polish_ports (
+      port_name STRING,
+      port_latitude FLOAT64,
+      port_longitude FLOAT64)
+''')
+```
+
+The `create_polish_ports_table` object is a lazy operation. Lazy means that calling `dataset.create_table`
+method, won't actually create a table. To run this lazy operation, you need to turn it into 
+a [job](./workflow-and-job.md#job) first, and then run it:
+
+```python
+create_polish_ports_table.to_job(id='create_ports_table').run()
+```
+
+Or put a job into a [workflow](./workflow-and-job.md#workflow) (note that there is not `run()` invocation):
+
+```python
+internationalports_workflow = Workflow(
+        workflow_id='internationalports',
+        definition=[
+                create_polish_ports_table.to_job(id='create_polish_ports_table'),
+        ],
+        schedule_interval='@once')
+```
+
+And then, run it using CLI:
+
+```shell script
+bf run --job internationalports.create_ports_table
+```
+
+Now, let us go through `DatasetConfig` and `Dataset` in detail.
+
+## Dataset Config
+
+`DatasetConfig` is a convenient extension to `Config` designed for workflows which use a `Dataset` object
```

Comment:
mamy to już opisane w configuration.md, na pewno podliknuj, może coś zdeduplikuj

Reply:
wyciągnięte z configuration.md